### PR TITLE
feat: add Christmas avatar and refactor avatar api

### PR DESCRIPTION
**Pull request by shadow2kk**

### CODE REVIEW

**Review by LucasM19 — APPROVED** [review]

File: apps/app/components/Settings/SettingAvatar.tsx [hunk 1]
```diff
@@ -90,9 +89,8 @@ export const SettingAvatar: React.FC<SettingAvatarProps> = ({ isSetup }) => {
   const api = useAPI()
   const toast = useToast()
   const navi = useNavigate()
-  const [avatars, setAvatars] = useAtom(avatarsAtom)
-
-  const [avatarSrc, setAvatarSrc] = useState('')
+  const [, setAvatars] = useAtom(avatarsAtom)
```
Comment (Codemasker) [inline apps/app/components/Settings/SettingAvatar.tsx:92]: `useUpdateAtom` should be used here

File: apps/home/constants/env.ts [hunk 2]
```diff
@@ -67,3 +67,5 @@ export const SUBSCRIBE_BUTTON_MIRROR_URL =
 export const SUBSCRIBE_BUTTON_BACKEND_URL =
   process.env.NEXT_PUBLIC_SUBSCRIBE_BACKEND_BUTTON_URL ||
   'https://subscribe.mail3.me'
+
+export const HOME_URL = process.env.NEXT_PUBLIC_HOME_URL || 'https://mail3.me'
```
Comment (Codemasker) [inline apps/home/constants/env.ts:71]: Why not use `location.origin` directly here. if it is in the `Nodejs`, You can use the `req.headers.host`

File: apps/home/pages/api/avatar/[address].ts [hunk 3]
```diff
@@ -2,7 +2,18 @@
 import { NextApiRequest, NextApiResponse } from 'next'
 import axios from 'axios'
 import { isPrimitiveEthAddress } from 'shared'
-import { SERVER_URL } from '../../../constants/env'
+
+import { SERVER_URL, HOME_URL } from '../../../constants/env'
+
+const DEFAULT_AVATAR_SRC =
```
Comment (Codemasker) [inline apps/home/pages/api/avatar/[address].ts:8]: It should be written in the environment variable.

File: apps/home/pages/api/avatar/[address].ts [hunk 4]
```diff
@@ -42,12 +53,18 @@ function handleSendFile(
 }
 
 async function address(req: NextApiRequest, res: NextApiResponse) {
+  const defaultAvatarUrl = `${HOME_URL}/avatar/${currentDefaultAvatar}.png`
```
Comment (Codemasker) [inline apps/home/pages/api/avatar/[address].ts:56]: If the default avatar is in the public directory, you can directly request `/public/avatar/normal.png`

File: packages/ui/src/Avatar/index.tsx [hunk 5]
```diff
@@ -37,6 +39,25 @@ const EMPTY_PLACE_HOLDER_SRC = 'empty_place_holder_image'
...
+  if (typeof img === 'string') return img
+  return img.src
+}
+
+const defaultAvatarMap = {
+  [DefaultAvatarType.Normal]: getImageSrc(PngAvatar),
+  [DefaultAvatarType.Christmas]: getImageSrc(PngAvatarChristmas),
+}
+
+const currentAvatarType = DefaultAvatarType.Christmas
```
Comment (Codemasker) [inline packages/ui/src/Avatar/index.tsx:57]: It should be controlled by environmental variables here.

**Review by Codemasker — CHANGES_REQUESTED** [review]

File: packages/ui/src/Avatar/index.tsx [hunk 6]
```diff
@@ -37,6 +39,25 @@ const EMPTY_PLACE_HOLDER_SRC = 'empty_place_holder_image'
 export const DEFAULT_AVATAR_SRC =
   'https://mail-public.s3.amazonaws.com/users/default_avatar.png'
 
+enum DefaultAvatarType {
+  Normal = 'normal',
+  Christmas = 'christmas',
+}
+
+const getImageSrc = (img: string | { src: string }) => {
```
Comment (Codemasker) [inline packages/ui/src/Avatar/index.tsx:47]: Just a line:
```suggestion
const getImageSrc = (img: string | { src: string }) => typeof img === 'string' ? img : img.src
```

**Review by Codemasker — CHANGES_REQUESTED** [review]

**Review by yuche — APPROVED** [review]

File: apps/home/pages/api/avatar/[address].ts [hunk 7]
```diff
@@ -1,8 +1,14 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
 import { NextApiRequest, NextApiResponse } from 'next'
 import axios from 'axios'
-import { isPrimitiveEthAddress } from 'shared'
-import { SERVER_URL } from '../../../constants/env'
+import { envStorage, isPrimitiveEthAddress } from 'shared'
+
+import { SERVER_URL, HOME_URL } from '../../../constants/env'
+// Constant will never change
```
Comment (Codemasker) [inline apps/home/pages/api/avatar/[address].ts:7]:
```suggestion
// Solve problems left over by history
```

**Review by Codemasker — CHANGES_REQUESTED** [review]

**Pull request merged**